### PR TITLE
docs(table): supplement string typing in table row key 

### LINKED ISSUE #3238: [BUG] - Table Row Key forces string typing.
### NextUI Version

2.4

### Describe the bug

The documentation about keys for the Table Rows is unclear and the typing implies that they are React.Key, which permits both numbers and strings. However, as far as I can tell from using the component, number ids/keys are converted to strings, and equality comparisons will subsequently fail. This was my workaround, to avoid the risk of breaking other parts of my projects API:

```
  const selectedSet = useMemo(() => {
    return forceString
      ? new Set<string>(selectedList.map((ident) => String(ident)))
      : new Set(selectedList);
  }, [selectedList, forceString]);

  const handleChange = useCallback(
    (selected: 'all' | Set<Key>) => {
      console.log(selected);
      const nextSelection =
        selected === 'all'
          ? [...idList]
          : ([...new Set(selected).values()] as U[]);
      dispatchSelected(nextSelection.sort());
    },
    [dispatchSelected, idList]
  );
  return { currentState, selectedSet, handleChange };
}

```

and then in the table:

```
  return (
    <Table
      aria-label="Example table with custom cells, pagination and sorting"
      isHeaderSticky
      bottomContent={bottomContent}
      bottomContentPlacement="outside"
      classNames={{
        wrapper: 'max-h-[382px]'
      }}
      selectedKeys={selectedSet} // this fails is the selectedSet is a set of numbers.
      selectionMode="multiple"
      sortDescriptor={sortDescriptor}
      topContent={topContent}
      topContentPlacement="outside"
      onSelectionChange={handleChange}
      onSortChange={setSortDescriptor}
    >
      <TableHeader columns={headerColumns}>
        {(column) => (
          <TableColumn
            key={column.uid}
            align={'start'}
            allowsSorting={column.sortable}
          >
            {column.name}
          </TableColumn>
        )}
      </TableHeader>
      <TableBody emptyContent={'No users found'} items={sortedItems}>
        {(item) => (
          <TableRow key={item.id}> // the id passed here is a number value, but apparently converted to a string by the TableRow 
            {(columnKey) => (
              <TableCell>{renderCell(item, columnKey)}</TableCell>
            )}
          </TableRow>
        )}
      </TableBody>
    </Table>
  );
```

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

I used the example Table from the NextUI docs:

https://nextui.org/docs/components/table#use-case-example

And ran into this bug when using entities that have numerical IDs.

### Expected behavior

I would expected the Table API to respect the data type of the provided keys, or for the documentation to otherwise be clear that it does not.

### Screenshots or Videos

_No response_

### Operating System Version

- Windows 10 Home V.22H2

### Browser

Chrome

### PR DESCRIPTION
**Pull request by awesome-pro**
Closes #3238

## 📝 Description

 Added the note to prevent possible bugs due to react key
```bash
React keys are automatically converted to strings. When using the Table component, ensure that all keys (e.g., row IDs) are treated as strings to avoid type mismatch issues.
```
## ⛳️ Current behavior (updates)

> Please describe the current behavior that you are modifying

## 🚀 New behavior

Screenshot 2024-07-10 at 5 56 14 PM

## 📝 Additional Information
 A very simple Docs Change

## Summary by CodeRabbit

- **New Features**
  - Added new event handlers for date picker component to improve focus and change events.
  - Updated popover placement from "right" to "top" for better usability.

- **Documentation**
  - Enhanced notes on key prop matching and React key conversion for the Table component to improve developer guidance.

### CODE REVIEW

**Comment by abhinandan-verma** [comment]
@wingkwong  I will solve all the unsolved bugs on  bugs in this same PR. And then I will change the title.

File: apps/docs/content/docs/components/table.mdx [hunk 1]
```diff
@@ -149,8 +149,11 @@ provide a default set of selected rows.
 
 <CodeDemo title="Single Row Selection" files={tableContent.singleSelection} />
 
-> **Note**: The value of the selected keys must match the key prop of the row.
+> **Note**:
+> - The value of the selected keys must match the `key` prop of the row.
+> - React keys are automatically converted to strings. When using the `Table` component, ensure that all keys (e.g., row IDs) are treated as strings to avoid type mismatch issues. For more information, refer to the [React documentation on keys](https://softwareengineering.stackexchange.com/questions/381268/why-are-react-keys-limited-to-strings)
```
Comment (wingkwong) [inline apps/docs/content/docs/components/table.mdx:154]: 1. `refer to the [React documentation on keys]`: That's not official React documentation.
2. `all keys (e.g., row IDs)`: change to `all row keys` instead.
3. Change the link `https://softwareengineering.stackexchange.com/questions/381268/why-are-react-keys-limited-to-strings/432948#432948` instead.

File: apps/docs/content/docs/components/table.mdx [hunk 2]
```diff
@@ -149,8 +149,11 @@ provide a default set of selected rows.
...
 
+This format makes it clear and easy to follow.
```
Comment (wingkwong) [inline apps/docs/content/docs/components/table.mdx:156]: please remove this line.

**Review by wingkwong — CHANGES_REQUESTED** [review]

**Comment by abhinandan-verma** [comment]
@wingkwong  Sir, there is some issue with `placement="right"` in docs for popover. This PR is just a workaround.
Many Popover are not working properly in docs.
I am working to resolve it.

File: apps/docs/content/docs/components/table.mdx (hunk 1 above)
Comment (wingkwong) [reply apps/docs/content/docs/components/table.mdx]: @abhinandan-verma You haven't addressed 1 - 3 at all.

**Comment by wingkwong** [comment]
> @wingkwong Sir, there is some issue with `placement="right"` in docs for popover. This PR is just a workaround. Many Popover are not working properly in docs. I am working to resolve it.

what is the issue? please don't mix multiple issues into one PR.

**Comment by wingkwong** [comment]
@abhinandan-verma Okay I got the popover issue now. Please remove your change from this PR and don't link those popover issues. That change will not be accepted

File: apps/docs/content/components/popover/usage.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ const App = `import {Popover, PopoverTrigger, PopoverContent, Button} from "@nex
 
 export default function App() {
   return (
-    <Popover placement="right">
+    <Popover placement="top">
```
Comment (wingkwong) [inline apps/docs/content/components/popover/usage.ts:5]: please undo it

File: apps/docs/content/docs/components/table.mdx [hunk 4]
```diff
@@ -149,7 +149,9 @@ provide a default set of selected rows.
 
 <CodeDemo title="Single Row Selection" files={tableContent.singleSelection} />
 
-> **Note**: The value of the selected keys must match the key prop of the row.
+> **Note**:
+> - The value of the selected keys must match the `key` prop of the row.
+> - React keys are automatically converted to strings. When using the `Table` component, ensure that all keys (e.g., row IDs) are treated as strings to avoid type mismatch issues. For more information, refer to the [React documentation on keys](https://reactjs.org/docs/lists-and-keys.html#keys)
```
Comment (wingkwong) [inline apps/docs/content/docs/components/table.mdx:154]: see my previous comment

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: apps/docs/content/docs/components/table.mdx [hunk 5]
```diff
@@ -149,7 +149,9 @@ provide a default set of selected rows.
 
 <CodeDemo title="Single Row Selection" files={tableContent.singleSelection} />
 
-> **Note**: The value of the selected keys must match the key prop of the row.
+> **Note**:
+> - The value of the selected keys must match the `key` prop of the row.
+> - React keys are automatically converted to strings. When using the `Table` component, ensure that all row keys (e.g., row IDs) are treated as strings to avoid type mismatch issues. For more information, [Refer Here](https://softwareengineering.stackexchange.com/questions/381268/why-are-react-keys-limited-to-strings/432948#432948)
```
Comment (wingkwong) [inline apps/docs/content/docs/components/table.mdx:154]: as mentioned before, please remove ` (e.g., row IDs) `

Comment (wingkwong) [inline apps/docs/content/docs/components/table.mdx:154]: only link the word `here` instead. Change it to lower case as well.

```
Refer [here]
```

**Review by wingkwong — CHANGES_REQUESTED** [review]

**Comment by abhinandan-verma** [comment]
@wingkwong  are you working on popover issues or I should work ?

**Comment by wingkwong** [comment]
You can investigate it but first finish this PR first. My comments are not resolved yet.

File: apps/docs/content/docs/components/table.mdx [hunk 6]
```diff
@@ -149,7 +149,9 @@ provide a default set of selected rows.
 
 <CodeDemo title="Single Row Selection" files={tableContent.singleSelection} />
 
-> **Note**: The value of the selected keys must match the key prop of the row.
+> **Note**:
+> - The value of the selected keys must match the `key` prop of the row.
+> - React keys are automatically converted to strings. When using the `Table` component, ensure that all row keys (e.g., row IDs) are treated as strings to avoid type mismatch issues. For more information, Refer [here](https://softwareengineering.stackexchange.com/questions/381268/why-are-react-keys-limited-to-strings/432948#432948)
```
Comment (wingkwong) [inline apps/docs/content/docs/components/table.mdx:154]: Remove `(e.g., row IDs)`. This is my third time to say that.

Comment (wingkwong) [inline apps/docs/content/docs/components/table.mdx:154]: 1. Refer -> refer
2. add a full stop at the end

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: apps/docs/content/docs/components/table.mdx [hunk 7]
```diff
@@ -149,7 +149,9 @@ provide a default set of selected rows.
 
 <CodeDemo title="Single Row Selection" files={tableContent.singleSelection} />
 
-> **Note**: The value of the selected keys must match the key prop of the row.
+> **Note**:
+> - The value of the selected keys must match the `key` prop of the row.
+> - React keys are automatically converted to strings. When using the `Table` component, ensure that all row keys are treated as strings to avoid type mismatch issues. For more information refer [here](https://softwareengineering.stackexchange.com/questions/381268/why-are-react-keys-limited-to-strings/432948#432948).
```
Comment (wingkwong) [inline apps/docs/content/docs/components/table.mdx:154]: i didn't ask for removing the comma. Please add it back.

**Review by wingkwong — CHANGES_REQUESTED** [review]
don't commit other changes unrelated to this PR.

**Review by wingkwong — CHANGES_REQUESTED** [review]
second thought. let's add it under `Usage` since it applies to all cases.

**Comment by wingkwong** [comment]
Closing - inactivity

**Pull request closed without merging**